### DISCUSSION TITLE
Battery degradation using Al-Dente

### DISCUSSION THREAD

**Discussion by ryders** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#770] [Q&A]
Folks, I started using AlDente ab 4 mths. Looking at my CoconutBattery online comparison, it appears the general state of health of my battery is getting worst and is not below that of the average battery out out there in a comparable machine (users which - I'm assuming most - do not use AlDente).

![Screen Shot 2022-09-14 at 00 14 30](https://user-images.githubusercontent.com/242383/190087840-5937a975-0e8c-460a-8a72-8dcbbe4f0bce.png)
https://www.coconut-flavour.com/ccbonline/index.php?bid=7cfa28895ae456f41bd8ee320e43ee7b

Does anyone have a different experience and do you mind sharing you settings details?

Roughly, I have my charge limit to 75%, with 40% Sailing Mode. I rarely charge to 100% and  will use battery power every couple of days for 4-6 hrs.. (down to 10%)

**Comment by MatthiasKerbl** [comment]
Hi @ryders,

If you rarely charge to 100% and discharge to 0% (which you should not do to keep your battery as healthy as possible) it is very likely that your battery calibration is off. Therefore, your battery capacity reading does not have much meaning. If you want to get a relatively good reading of your battery capacity you need to cycle your battery a couple of times between 0% and 100%. Afterward, compare your capacity again.

Cheers,
Matthias

**Reply by ryders** [reply]
Hi @MatthiasKerbl thanks for chiming in. I definitely do calibration every 2 weeks, and I normally use about 8-10hrs on the battery each week, so definitely not "always plugged in".. but mostly.

**Comment by ryders** [comment]
Hi @MatthiasKerbl 

I realise this may not be right place for community feedback.

I just got a new battery installed (for various reasons not related to Al-Dente) and I'm now wondering where you know of where I could find evidence that battery health "degrades less slowly with Al-Dente".

I understand it's not obvious and settings + usage patterns will vary between people, but I'm trying to find at least one case where Al-Dente CLEARLY made things better..

A mate of mine using Al-Dente sent me this https://www.coconut-flavour.com/ccbonline/index.php?bid=a0fef974bf85bdf49792e368cadba012
very similar to mine, above.


Any thoughts?

**Reply by MatthiasKerbl** [reply]
Hi @ryders,

So far, we have not done a scientific study with AlDente since we currently do not have the financial resources or the time capacity to do that properly.
The concept of AlDente is based on the following Argument: Scientific studies indicate a couple of things that result in faster battery degradation (heat, staying at high or low SoC, high charging- or discharging rate, a high number of cycles,...). Therefore, with AlDente, we try to give our users the tools to counteract these damaging scenarios while still being able to conveniently use their MacBooks.
If you are interested in studies of battery degradation, you can find them in scientific literature databases. Some studies to start with are called:

- Cycle life testing and modeling of graphite/LiCoO2 cells under different state of charge ranges by Saurabh Saxena, Christopher Hendricks, Michael Pecht
- Extending Battery Lifetime by Avoiding High SOC by Evelina Wikner and Torbjörn Thiringer
- Strategies to limit degradation and maximize Li-ion battery service lifetime - Critical review and guidance for stakeholders by Maxwell Woodya, Maryam Arbabzadeha, Geoffrey M. Lewisa, Gregory A. Keoleiana, Anna Stefanopouloub

Hope that helps.

Cheers,
Matthias

**Comment by ValeriiSohlaiev** [comment]
> Folks, I started using AlDente ab 4 mths. Looking at my CoconutBattery online comparison, it appears the general state of health of my battery is getting worst and is not below that of the average battery out out there in a comparable machine (users which - I'm assuming most - do not use AlDente).
> 
> ![Screen Shot 2022-09-14 at 00 14 30](https://user-images.githubusercontent.com/242383/190087840-5937a975-0e8c-460a-8a72-8dcbbe4f0bce.png) https://www.coconut-flavour.com/ccbonline/index.php?bid=7cfa28895ae456f41bd8ee320e43ee7b
> 
> Does anyone have a different experience and do you mind sharing you settings details?
> 
> Roughly, I have my charge limit to 75%, with 40% Sailing Mode. I rarely charge to 100% and will use battery power every couple of days for 4-6 hrs.. (down to 10%)

MacBook Pro (14-inch, 2021)

I immediately installed the AlDente app after purchase. And I just want to understand why the program doesn’t help me.

   Health Information:
   Cycle Count: 100
   Condition: Normal
   Maximum Capacity: 91%

**Reply by ryders** [reply]
Hi @ValeriiSohlaiev 

I'm not sure I understand your question, and I don't believe you're trying to answer mine 😝 Your battery sure seems healthy, but not healthier than users without AlDente. Right? 

I have used Al-Dente for quite some time now (8mths), Pro version also, and here's my understand of how it helps you in a few points. (and for the benefit of others)

1 - some suggest that "two boys cannot be better than Apple" at managing power;
2 - many point out that Apple has a history of releasing half-baked software also;
3 - the Machine Learning "Optimised charging" will take time to properly learn all usage patterns and properly apply to your usage.. and still unclear how it adapts to those not "vastly unclear" charge+usage patterns.
4a - Ultimately, batteries are consumables and will degrade no matter what. The question is whether you care about / enjoy fiddling this AlDente to optimise your own setup, or whether you care more about your time spent doing something else 😆

4b - AlDente just gives you the flexibility to "force the charge settings" you think are more accurate. (I read somewhere that the code is actually not "bypassing" the charge controller, but really "cleanly" instructing the OS on when to and when not to charge). So ultimately, yes, Al-Dente will allow you to muck around with your settings, and possibly throw calibration out of whack.


my 2 cents;
- the heat protection setting in Al-Dente is valuable: heat kills batteries
- calibrating once per month, up to once per fortnight (is always plugged in) is good hygiene, if you don't you may have to calibrate a few times to get back to an accurate calibration
- I've read many suggesting to disconnect the charger overnight as a habit.


I'm almost always connected and, when I use the battery (total ~1 full cycle per week) it's usually only for a few (3-5) hours, so I don't need 100% battery, The rare times I do need 100%, I use the "top up" button beforehand. I've been experimenting with charge capped at 60, 65, 70%... with a sail mode of (respectively) 20, 30, 40%.. so that I'm always with a constant +/- 10, 15 or 30% from the "50% mark" which I believe to be the least damaging for the battery chemistry.

@MatthiasKerbl curious to hear your thoughts on that last paragraphs? my usage and chemistry?

Hope this helps @ValeriiSohlaiev or anyone

**Reply by ValeriiSohlaiev** [reply]
I wanted to say that this program is absolutely useless. It didn't help in my case.

**Reply by DamonHD** [reply]
Why would you come here to say such an unconstructive and unkind and untrue thing?

It may not help you, but it is clearly NOT "absolutely useless".  It does what I want from it, for example.

Rgds

Damon

**Reply by ValeriiSohlaiev** [reply]
I told the truth. By using this app I was hoping to keep my battery alive longer. I used the pro version and paid for it and got nothing. In less than a year of use and a hundred cycles, I got 91% battery and life. If I'm wrong or if you know how to fix this please let me know.

**Reply by MatthiasKerbl** [reply]
Hi @ryders,

Here is my opinion on some of your statements:

> 1 - some suggest that "two boys cannot be better than Apple" at managing power

In my opinion, it is wrong to think that Apple wants to offer the best battery longevity for its MacBooks. First of all, they have a huge conflict of interest because they mostly profit off of fast-degrading batteries. Secondly, even if they did not have this conflict of interest, in my opinion, Apple is more focused on delivering the most convenient user experience for the main majority of its users.

> 4a - Ultimately, batteries are consumables and will degrade no matter what. The question is whether you care about / enjoy fiddling this AlDente to optimise your own setup, or whether you care more about your time spent doing something else 😆

Good point. Some people just do not care about their MacBook battery longevity or do not want to educate themselves about it. However, the question some people have to ask themselves is, if their usage pattern might result in a broken battery after a year which leads to spending around 200 USD on a new battery, or spending around 24 USD on AlDente Pro and potentially avoiding the broken battery after a year.

> 4b - AlDente just gives you the flexibility to "force the charge settings" you think are more accurate. 
Correct.

> (I read somewhere that the code is actually not "bypassing" the charge controller, but really "cleanly" instructing the OS on when to and when not to charge).

Correct. We want to give users the tools to control the charging behavior of their MacBooks the way they want. Also, see my reply above for more details about that.

> So ultimately, yes, Al-Dente will allow you to muck around with your settings, and possibly throw calibration out of whack.

Correct. However, even if a miscalibration occurs, this is only a software issue and does not result in any "real" damage and can be reversed with the Calibration Mode of AlDente or manually cycling the battery between 0% and 100%.

> I've read many suggesting to disconnect the charger overnight as a habit.

Might be a good idea.

> I'm almost always connected and, when I use the battery (total ~1 full cycle per week) it's usually only for a few (3-5) hours, so I don't need 100% battery, The rare times I do need 100%, I use the "top up" button beforehand. I've been experimenting with charge capped at 60, 65, 70%... with a sail mode of (respectively) 20, 30, 40%.. so that I'm always with a constant +/- 10, 15 or 30% from the "50% mark" which I believe to be the least damaging for the battery chemistry.

Sounds pretty good to me. I would probably also calibrate the battery once a week or once every two weeks. Also, the "50% mark" might be a good rule of thumb but I have not really seen enough data that suggests that it is the perfect battery level for longevity.

Again, hope that helps.

Cheers,
Matthias

**Reply by MatthiasKerbl** [reply]
Hi @ValeriiSohlaiev,

I am sorry that you feel disappointed by AlDente. However, in my opinion, there are a bunch of misconceptions and misunderstandings about batteries and AlDente. We will try to improve our communication about that.  I have also started writing a couple of blog posts about those topics. Unfortunately, they are not finished yet and I do not know when I will be able to finish them.

Maybe these quick answers to common questions will help:

- "Will AlDente stop battery degradation completely?"
No, AlDente tries to slow it down. Every Battery degrades over time.

- "Why did my battery lose 8% capacity in just 6 months with AlDente installed?"
Most Batteries lose between 5%-10% in the first 6-12 months due to various mechanisms in the battery. Unfortunately, I have not seen anything that can be done about that. However, AlDente provides the tools to slow down the degradation process after the first phase.

- "How much battery capacity will AlDente save?"
Unfortunately, this is impossible to answer because it is different for every single user because it depends on many different factors and usage behavior. Furthermore, every battery is different and will have a different result even under the same conditions. However, scientific studies show a huge difference in battery degradation for different charge limits and other battery-damaging usages.

In my opinion, losing 9% battery capacity in the first phase after 100 cycles and 6-12 months might be a little bit more than average with AlDente but is nevertheless not extremely unlikely. Also, checking the battery capacity only once without calibrating it does not offer a lot of information about what the real battery capacity is.

Hope that clarifies a couple of things.

Cheers,
Matthias

**Reply by MatthiasKerbl** [reply]
Also @ryders and @DamonHD, thanks for trying to help other users.
This is much appreaciated!

**Reply by jotala** [reply]
> > So ultimately, yes, Al-Dente will allow you to muck around with your settings, and possibly throw calibration out of whack.
> 
> Correct. However, even if a miscalibration occurs, this is only a software issue and does not result in any "real" damage and can be reversed with the Calibration Mode of AlDente or manually cycling the battery between 0% and 100%.
> 
> >

@ryders 
I'm using Al Dente since 2021-11, it's true if you don't do a full cycle Aldente throw calibration out. But my battery is safer and that's the point. With my 16 M1, I never have used it for more than 30%. So I did keep my charge limit at 60%.
Look how much my battery has improved this month when I did half drain to 20% and charged fully.

Look how much my battery has improve this month when I did half drain to 20% and charged full.

https://www.coconut-flavour.com/ccbonline/index.php?bid=80182aa4db16afae4d0c63672febc06e

<img width="1032" alt="Screenshot 2022-12-22 at 18 04 33" src="https://user-images.githubusercontent.com/112975623/209227018-134bb575-d6d2-4886-bd00-5e53e7250c14.png">

@MatthiasKerbl 

Suggestion
Option to drain it to 0% when calibrating. You could add a field to choose the minimum %.
I have the impression it's good for the battery every 6 months drain every drop of the battery and do a partial calibration every month or two.

I have some questions:

1 - I bought the Pro version. Since I'm a Setapp subscriber and I love your App. (every penny counts) Is there any difference between Setapp and bought from you?
2 - On System Settings App - I have "Optimized Battery Charging" Off, and "Low power mode" "Only on Battery". The latter I haven't seen them on your site. Does it matter?
3 - While using Al Dente do the Mac Index Spotlight, OCR, and the Photos App do the face recognition normally?
Thank you

**Reply by MatthiasKerbl** [reply]
Hi @jotala,

1. No, there is no difference between the Setapp version or the version you buy through our website.
2. So far, we have not experienced any affect of the Low power mode on AlDente.
3. As far as I know, all of them work as usual.

Cheers,
Matthias

**Comment by angelyordanov** [comment]
I would like to give my two cents on the topic.

I'm using AlDente Pro exclusively with my Macbook Pro 16" and 95% of the time it is connected to an Apple Studio Display and thus always on power.

I do see larger degradation from my previous macbooks which I manually managed by plugging/unplugging the power cable. But I think this is largely due to AlDent messing up with the calibration in a big way.  After a few very deep (<10%) discharges with full 100% charge and hold the degradation went from 89% to 96%. And then in a few days it went down to 92% again.

https://www.coconut-flavour.com/ccbonline/index.php?bid=3dcf99e47959916178524122fae1d9ff
<img width="1031" alt="Screenshot 2022-12-12 at 14 19 39" src="https://user-images.githubusercontent.com/1524014/207043780-311db288-ae84-4ced-8938-494945317997.png">
<img width="435" alt="Screenshot 2022-12-12 at 13 53 04" src="https://user-images.githubusercontent.com/1524014/207044002-229cf940-8f29-48a8-a68c-b1889f9de1c5.png">

In my view if you are going to be connected to power most of the time the app is a must have, but if you can cycle charge/discharge with the power cable and your daily usage fits within 1 cycle the deg will be comparable.

@MatthiasKerbl a few suggestions. I like cycling the battery with AlDent too and the new feature "Automatic Discharge" helps for that as upon unplugging/plugging the "discharge" wont be reset, but it would be even better if there is a "mode" where one specifies upper and lower limit and the app cycles continuously upper -> lower -> upper and so on.

**Reply by ryders** [reply]
@angelyordanov Thanks for chiming in, mate 👍 While I can't say that I have a studio display -- I also have a similar pattern where I'm connected 90-95% of the time.

I'm pretty sure the pattern you're suggesting is not _AlDente messing up with calibration in a big way_, but instead something along the lines of battery chemistry or memory or rubbish BMS (ahem, or better adapted to most users?) from Apple and in essence, the battery/bms doesn't have any indication it can access that extra 6-7% until you forcefully prove it can by calibrating. In other words, any charge/discharge limitation seems to have that effect on a battery.

@MatthiasKerbl you probably know more about battery chemistry -- feel free to add your 2c -- but my understanding is that Lithium-ion doesn't have a clear "memory effect" as other chemistries, however cells degrade differently and this impact overall "working voltage", or the (perceived?) ability to deliver a voltage on demand. In other words, the battery management, when "probing" or "measuring" cell delivery, perceives that some cells are not able to store as much energy because they "discharge too fast" under load.

This said, I have had a new battery installed 2 months ago, and I have been using AlDente since new, my settings are 80% Charge Limit, with 55% discharge (will sail to 25%). I've been unplugging the laptop at least once a week, consuming about 30-50% of the battery, and so far, I'm trending above average:

![Screenshot 2022-12-13 at 11 31 04 am](https://user-images.githubusercontent.com/242383/207298425-807f1057-1eb1-43a1-aab1-ba423c371de2.png)

https://www.coconut-flavour.com/ccbonline/index.php?bid=8ae4cd3b11ef7f637e503235da35670c

**Reply by MatthiasKerbl** [reply]
Hi @angelyordanov,

I will try to share my opinion about some thoughts you shared.

>... but if you can cycle charge/discharge with the power cable and your daily usage fits within 1 cycle the deg will be comparable.

In my opinion, this is not correct. Even though battery degradation is a complex topic and depends on many factors, it is clear that battery cycle count is one important factor in the long term. Therefore, the slower the cycle count increases, the better. If you cycle the battery daily, the cycle count will definitely increase faster than when doing one full cycle once a week. As a result, the battery will theoretically also degrade much faster than with option two.

> @MatthiasKerbl a few suggestions. I like cycling the battery with AlDent too and the new feature "Automatic Discharge" helps for that as upon unplugging/plugging the "discharge" wont be reset, but it would be even better if there is a "mode" where one specifies upper and lower limit and the app cycles continuously upper -> lower -> upper and so on.

We have actively decided not to include a feature like that. The reason for that is similar to the above. Battery research strongly indicates that leaving a battery at a somewhat healthy percentage of for example 80% instead of cycling it through a certain range, results in much slower battery degradation. To keep the battery calibrated, we implemented the Calibration Mode, which will perform an x%-100%-15%-100%-x% cycle.

Hope that clarifies some things.

Cheers,
Matthias
